server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -110,22 +110,36 @@ export default class PreviousWorkExperienceTypesUpdateController extends Previou
 
     req.session.previousWorkExperienceTypesForm = undefined
 
-    const pageFlowQueue = this.buildPageFlowQueue(workExperienceTypesToShowDetailsFormFor, prisonNumber)
+    const pageFlowQueue = this.buildPageFlowQueue(
+      workExperienceTypesToShowDetailsFormFor,
+      prisonNumber,
+      this.previousPageWasCheckYourAnswers(req),
+      req.session.pageFlowQueue,
+    )
+
     req.session.pageFlowQueue = pageFlowQueue
     return res.redirect(getNextPage(pageFlowQueue))
   }
 
   buildPageFlowQueue = (
     previousWorkExperienceTypes: Array<TypeOfWorkExperienceValue>,
     prisonNumber: string,
+    fromCheckAnswers: boolean,
+    currentPageFlow?: PageFlow,
   ): PageFlow => {
-    const previousWorkExperienceTypesPageUrl = `/prisoners/${prisonNumber}/induction/previous-work-experience`
-    const pageUrls = [
-      previousWorkExperienceTypesPageUrl,
-      ...previousWorkExperienceTypes.map(
-        workType => `/prisoners/${prisonNumber}/induction/previous-work-experience/${workType.toLowerCase()}`,
-      ),
-    ]
+    const nextPages = previousWorkExperienceTypes.map(
+      workType => `/prisoners/${prisonNumber}/induction/previous-work-experience/${workType.toLowerCase()}`,
+    )
+
+    // If not coming from check answers and on an existing page flow, append to the current flow
+    if (currentPageFlow && !fromCheckAnswers) {
+      return appendPagesFromCurrentPage(currentPageFlow, nextPages)
+    }
+    const pageUrls = [`/prisoners/${prisonNumber}/induction/previous-work-experience`, ...nextPages]
+    // If coming from check answers, redirect back at end of journey flow
+    if (fromCheckAnswers) {
+      pageUrls.push(`/prisoners/${prisonNumber}/induction/check-your-answers`)
```

Comment:
If the user got here from the check answers page change link instead of the "Worked Before" page, then add redirect back to the check your answers page at the end of the flow.